src/shared/types.ts

```diff
@@ -6,3 +8,40 @@ export interface IframePhone {
   initialize: () => void;
   disconnect: () => void;
 }
+
+export type WrapperType = "scaler" | "carousel" | "scaffolding";
+
+// Note that TS interfaces should match JSON schema. Currently there's no way to generate one from the other.
+// TS interfaces are not available in runtime in contrast to JSON schema.
+export interface IAuthoredState extends IAuthoringInteractiveMetadata {
```

Comment:
Each of the interactives define their own IAuthoredState and IInteractiveState types that are specific to them. 
I realize that the types for Carousel and Scaffolded question are almost identical, so it is tempting to combine them like this. But they should be kept separate for clarity and maintainability.